platform.in

```diff
@@ -67,6 +67,20 @@
 #==============================================================================
 # ---- Build policy macros.
 #
+
+%__buildroot_clean %{__rm} -rf "%{buildroot}"} \
+    %{__mkdir_p} "%{dirname:%{buildroot}}"\
+    %{__mkdir} "%{buildroot}"\
+%{nil}
+
+#---------------------------------------------------------------------
+#	Expanded at beginning of %install scriptlet.
+#
+
+%__spec_install_pre %{___build_pre}\
+   %{?buildroot:%{?__buildroot_clean}}\
+%{nil}
+
 #---------------------------------------------------------------------
```

Comment:
The test whether buildroot is defined is still there.
Oh and one more thing: I think the name "clean" sends the wrong message here, it's easily associated with the %clean section when that's not the case at all. The simplest way to eliminate the naming question would be just putting this all into %__spec_install_pre directly (ie without making it a function). It's not like this is going to be useful in any other section.

Reply:
Fixed.